src/firestarter/Environment/X86/Payload/AVX512_AMX_Payload.cpp

```diff
@@ -0,0 +1,592 @@
+/******************************************************************************
+ * FIRESTARTER - A Processor Stress Test Utility
+ * Copyright (C) 2020 TU Dresden, Center for Information Services and High
+ * Performance Computing
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/\>.
+ *
+ * Contact: daniel.hackenberg@tu-dresden.de
+ *****************************************************************************/
+#include <asm/prctl.h>        /* Definition of ARCH_* constants */
+#include <firestarter/Environment/X86/Payload/AVX512_AMX_Payload.hpp>
+#include <sys/syscall.h>
+#include <immintrin.h>
+
+#define XFEATURE_XTILECFG   17
+#define XFEATURE_XTILEDATA  18
+#define XFEATURE_MASK_XTILECFG  (1 << XFEATURE_XTILECFG)
+#define XFEATURE_MASK_XTILEDATA (1 << XFEATURE_XTILEDATA)
+#define XFEATURE_MASK_XTILE (XFEATURE_MASK_XTILECFG | XFEATURE_MASK_XTILEDATA)
+
+#define ARCH_GET_XCOMP_PERM 0x1022
+#define ARCH_REQ_XCOMP_PERM 0x1023
+
+#define MAX 1024
+#define MAX_ROWS 16
+#define MAX_COLS 64
+
+
+using namespace firestarter::environment::x86::payload;
+using namespace asmjit;
+using namespace asmjit::x86;
+
+// Define struct that is used as config and loaded through ldtilecfg()
+typedef struct __tile_config
+{
+  uint8_t palette_id;
+  uint8_t start_row;
+  uint8_t reserved_0[14];
+  uint16_t colsb[16];
+  uint8_t rows[16];
+} __tilecfg;
+
+void create_AMX_config(__tilecfg *tileinfo);
+void request_permission();
+void init_buffer_rand(uintptr_t buf1, uintptr_t buf2);
+
```

Comment:
please define them as static functions inside the AVX512_AMX_Payload class or inside a new class handling the amx permissions and configs

Reply:
Done